modules/core/src/stat.cpp

```diff
@@ -4150,7 +4150,10 @@ double cv::PSNR(InputArray _src1, InputArray _src2)
 {
     CV_INSTRUMENT_REGION()
 
-    CV_Assert( _src1.depth() == CV_8U );
+    //Input arrays must have depth CV_8U and be of identical size
+    CV_Assert( _src1.depth() == CV_8U && _src2.depth() == CV_8U );
+    CV_Assert( _src1.rows() == _src2.rows() && _src1.cols() == _src2.cols() && _src1.dims() == _src2.dims() );
```

Comment:
There is no need in the second assert. The function calls cv::norm() which does the same check. When writing high-level functions that are based on lower-level functions, we should rely on the checks that those lower-level functions do, otherwise the code will be flooded with duplicate assert's. The first assert still makes sense because the normalization in PSNR assumes that the arrays are 8-bit.